atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestV2Table.java

```diff
@@ -0,0 +1,271 @@
+                          SchemaApiTestTable.SchemaApiTestRowResult.of(entry);
+                     resultsMap.put(resultEntry.getRowName().getComponent1(), resultEntry.getColumn1());
+                });
+        return resultsMap;
+    }
+
+    /**
+     * Returns the value for column Column2 and specified row components. */
+    public Optional<String> getColumn2(String component1) {
+        SchemaApiTestTable.SchemaApiTestRow row = SchemaApiTestTable.SchemaApiTestRow.of(component1);
+        byte[] bytes = row.persistToBytes();
+        ColumnSelection colSelection = 
+                ColumnSelection.create(ImmutableList.of(PtBytes.toCachedBytes("d")));
+        RowResult<byte[]> rowResult = t.getRows(tableRef, ImmutableSet.of(bytes), colSelection).get(bytes);
+        if (rowResult == null) {
+             return Optional.empty();
+        } else {
+             return Optional.of(SchemaApiTestTable.SchemaApiTestRowResult.of(rowResult).getColumn2());
+        }
+    }
+
+    /**
+     * Returns a mapping from row keys to value at column Column2.
+     * As the values are all loaded in memory, do not use for large amounts of data.
+     * If the column does not exist for a key, the entry will be omitted from the map. */
+    public Map<String, String> getColumn2(Iterable<String> rowKeys) {
+        ColumnSelection colSelection = 
+                 ColumnSelection.create(ImmutableList.of(PtBytes.toCachedBytes("d")));
+        List<SchemaApiTestTable.SchemaApiTestRow> rows = Lists
+                .newArrayList(rowKeys)
+                .stream()
+                .map(SchemaApiTestTable.SchemaApiTestRow::of)
+                .collect(Collectors.toList());
+        SortedMap<byte[], RowResult<byte[]>> results = t.getRows(tableRef, Persistables.persistAll(rows), colSelection);
+        return results
+                .values()
+                .stream()
+                .map(entry -> SchemaApiTestTable.SchemaApiTestRowResult.of(entry))
+                .collect(Collectors.toMap(
+                     entry -> entry.getRowName().getComponent1(), 
+                     SchemaApiTestTable.SchemaApiTestRowResult::getColumn2));
+    }
+
+    /**
+     * Returns a mapping from all the row keys to their value at column Column2
+     * (if that column exists for the row-key). As the values are all loaded in memory,
+     * do not use for large amounts of data. */
+    public Map<String, String> getAllColumn2() {
+        ColumnSelection colSelection = 
+                ColumnSelection.create(ImmutableList.of(PtBytes.toCachedBytes("d")));
+        return getSmallRowRangeColumn2(RangeRequest.all());
+    }
+
+    /**
+     * Returns a mapping from all the row keys in a rangeRequest to their value at column Column2
+     * (if that column exists). As the values are all loaded in memory, do not use for large amounts of data.  */
+    public Map<String, String> getSmallRowRangeColumn2(RangeRequest rangeRequest) {
```

Comment:
Given that there is only one row component here, it might be nice to have a method `getSmallRangeColumn2(String component1StartIndlusive, String component1EndExclusive, int limit)` (which could just call the above one)

Reply:
also, let's make this return a `LinkedHashMap` and document that it has items in the correct order